operators/src/source/ogr_source.rs

```diff
@@ -0,0 +1,2856 @@
+                OgrSourceProcessor::new(
+                    self.state.dataset_information.clone(),
+                    self.state.default_geometry.try_into_geometry()?,
+                )
+                .boxed(),
+            ),
+            VectorDataType::MultiLineString => TypedVectorQueryProcessor::MultiLineString(
+                OgrSourceProcessor::new(
+                    self.state.dataset_information.clone(),
+                    self.state.default_geometry.try_into_geometry()?,
+                )
+                .boxed(),
+            ),
+            VectorDataType::MultiPolygon => TypedVectorQueryProcessor::MultiPolygon(
+                OgrSourceProcessor::new(
+                    self.state.dataset_information.clone(),
+                    self.state.default_geometry.try_into_geometry()?,
+                )
+                .boxed(),
+            ),
+        })
+    }
+}
+
+pub struct OgrSourceProcessor<G>
+where
+    G: Geometry + ArrowTyped,
+{
+    dataset_information: Arc<OgrSourceDataset>,
+    default_geometry: Option<G>,
+    _collection_type: PhantomData<FeatureCollection<G>>,
+}
+
+impl<G> OgrSourceProcessor<G>
+where
+    G: Geometry + ArrowTyped,
+{
+    pub fn new(dataset_information: Arc<OgrSourceDataset>, default_geometry: Option<G>) -> Self {
+        Self {
+            dataset_information,
+            default_geometry,
+            _collection_type: Default::default(),
+        }
+    }
+}
+
+impl<G> QueryProcessor for OgrSourceProcessor<G>
+where
+    G: Geometry + ArrowTyped + 'static + std::marker::Unpin + TryFromOgrGeometry,
+    FeatureCollectionRowBuilder<G>: FeatureCollectionBuilderGeometryHandler<G>,
+{
+    type Output = FeatureCollection<G>;
+    fn query(&self, query: QueryRectangle, ctx: QueryContext) -> BoxStream<Result<Self::Output>> {
+        OgrSourceQuery::new(
+            self.dataset_information.clone(),
+            self.default_geometry.clone(),
+            query,
+            ctx.chunk_byte_size,
+        )
+        .boxed()
+    }
+}
+
+pub struct OgrSourceQuery<G>
+where
+    G: Geometry + ArrowTyped,
+{
+    worker_thread_is_idle: bool,
+    worker_thread_terminated: bool,
+    poll_result_receiver: Receiver<Option<Result<FeatureCollection<G>>>>,
+    work_query_sender: SyncSender<WorkQuery>,
+    _geometry_type: PhantomData<G>,
+}
+
+struct WorkQuery {
+    waker: Waker,
+}
+
+impl<G> OgrSourceQuery<G>
+where
+    G: Geometry + ArrowTyped + 'static + TryFromOgrGeometry,
+    FeatureCollectionRowBuilder<G>: FeatureCollectionBuilderGeometryHandler<G>,
+{
+    pub fn new(
+        dataset_information: Arc<OgrSourceDataset>,
+        default_geometry: Option<G>,
+        query_rectangle: QueryRectangle,
+        chunk_byte_size: usize,
+    ) -> Self {
+        let (poll_result_sender, poll_result_receiver) = mpsc::sync_channel(2);
+        let (work_query_sender, work_query_receiver) = mpsc::sync_channel(1);
+
+        spawn_blocking(move || {
```

Comment:
Well, there is some overhead for every stream creation. The complication is that I cannot send any GDAL type, so I have to init it very early.

Reply:
it's true that there is some overhead, but we should minimize it. Please add a comment that you start the thread here already because GDAL types are not `send`

Reply:
b8a82ce